aws/resource_aws_eip_test.go

```diff
@@ -306,6 +306,35 @@ func TestAccAWSEIP_tags(t *testing.T) {
 	})
 }
 
+func TestAccAWSEIP_PublicIpv4Pool(t *testing.T) {
+	var conf ec2.Address
+	resourceName := "aws_eip.bar"
+	envPoolName := fmt.Sprintf("%s", os.Getenv("AWS_EC2_EIP_PUBLIC_IPV4_POOL"))
+	poolName := ""
+	if envPoolName == "" {
+		poolName = "amazon"
+	} else {
+		poolName = envPoolName
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_eip.bar",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSEIPDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEIPConfig_PublicIpv4Pool(poolName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEIPExists(resourceName, &conf),
+					testAccCheckAWSEIPAttributes(&conf),
+					resource.TestCheckResourceAttr(resourceName, "public_ipv4_pool", poolName),
```

Comment:
We should verify that the "basic" tests above have this attribute set to `"amazon"` for EC2-Classic and VPC EIPs.

Reply:
As I mentioned above EC2-Classic does not allow to use `public_ipv4_pool` attribute.